web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +193,456 @@ describe("<TaskHeader />", () => {
     });
   });
 
+  describe("tax registration modal", () => {
+    it("opens when tax task is marked complete", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["START"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "START" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      expect(screen.getByText(Config.taxRegistrationModal.title)).toBeInTheDocument();
+    });
+
+    it("doesn't update the task status when closed", async () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByTestId("modal-button-secondary"));
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.NOT_STARTED)).toBeInTheDocument();
+      });
+    });
+
+    it("updates the task progress when the modal is saved", async () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByTestId("modal-button-primary"));
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.COMPLETED)).toBeInTheDocument();
```

Comment:
the `within` technique is a good idea here, and it'd be nice if we refactored the rest of this test file to use it, instead of how the other tests use `[0]` or `getAllByText` to handle this

Reply:
not as part of this story but as a follow-up refactor